src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -1109,6 +1109,67 @@ inline std::istream& operator>>(std::istream& is, Affinity& affinity) {
  */
 static constexpr Property<Affinity> affinity{"AFFINITY"};
 
+/**
+ * @brief The name for setting to execute in bfloat16 precision whenever it is possible
+ * @ingroup ov_runtime_cpp_prop_api
+ *
+ * This option let plugin know to downscale the precision where it see performance benefits from
+ * bfloat16 execution
+ * Such option do not guarantee accuracy of the network, the accuracy in this mode should be
+ * verified separately by the user and basing on performance and accuracy results it should be
+ * user's decision to use this option or not to use
+ */
+static constexpr Property<bool> enforce_bf16{"ENFORCE_BF16"};
+
+/**
+ * @brief Enum to define possible snippets mode hints
+ * @ingroup ov_runtime_cpp_prop_api
+ */
+enum class SnippetsMode {
+    ENABLE = 0,           //!<  Enable
+    IGNORE_CALLBACK = 1,  //!<  Ignore callback
+    DISABLE = 2,          //!<  Disable
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const SnippetsMode& mode) {
+    switch (mode) {
+    case SnippetsMode::ENABLE:
+        return os << "ENABLE";
+    case SnippetsMode::IGNORE_CALLBACK:
+        return os << "IGNORE_CALLBACK";
+    case SnippetsMode::DISABLE:
+        return os << "DISABLE";
+    default:
+        OPENVINO_THROW("Unsupported snippets mode value");
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, SnippetsMode& mode) {
+    std::string str;
+    is >> str;
+    if (str == "ENABLE") {
+        mode = SnippetsMode::ENABLE;
+    } else if (str == "IGNORE_CALLBACK") {
+        mode = SnippetsMode::IGNORE_CALLBACK;
+    } else if (str == "DISABLE") {
+        mode = SnippetsMode::DISABLE;
+    } else {
+        OPENVINO_THROW("Unsupported snippets mode: ", str);
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief Defines Snippets tokenization mode
+ *      @param ENABLE - default pipeline
+ *      @param IGNORE_CALLBACK - disable the Snippets markup transformation and tokenization callback
+ *      @param DISABLE - turn off the Snippets
+ * @ingroup ie_dev_api_plugin_api
+ */
+static constexpr Property<size_t, PropertyMutability::RW> snippets_mode{"SNIPPETS_MODE"};
```

Comment:
I suppose we should not extend public API, because these properties are internal API

Reply:
> SNIPPETS_MODE is cpu plugin property?

yes, they are not used in any other plugin.
Anyway, you should not introduce any new public API in this PR

Reply:
Seems `enforce_bf16` is public property, which is needed but not in public, where to put it? @ilya-lavrenov

Reply:
I see it in public only for old API

Reply:
yes, `ov::hint::inference_precision` can cover `ENFORCE_BF16` functionality, but there is minor difference - `ENFORCE_BF16` is a quit separated property, there should be specified use case? and about 30 test files contains this property's tests. 

Anyhow,  could you confirm whether we don't migrate  `ENFORCE_BF16`  to new API?
If not migrate `ENFORCE_BF16`,  we  can use `ov::hint::inference_precision`  to replace  `ENFORCE_BF16` in new API.

Reply:
I can confirm that `ENFORCE_BF16` is not required in public API.

Looks like it's used in tests and that is why it's added to plugin API.
@maxnick do you think we still `ENFORCE_BF16` for tests purposes or we can rely on public API and use inference_precision hint?

Reply:
@EgorDuplensky could you give some comments on `ENFORCE_BF16` question?

Reply:
@riverlijunjie  `ENFORCE_BF16` should be fully replaced with `ov::hint::inference_precision`. Lets not propogare this old property into API 2.0

Reply:
ok, I have updated to not introduce it to API 2.0.